### PR TITLE
Added getacqinfo / getfileinfo functions for HFB acquisitions.

### PR DESCRIPTION
**Pull request by james-s-willis**

### CODE REVIEW

**Comment by james-s-willis** [comment]
@ketiltrout is there a way to test this before merging?

File: alpenhorn/auto_import.py [hunk 1]
```diff
@@ -470,6 +533,22 @@ def _import_file(node, root, acq_name, file_name):
...
                 )
                 di.CorrAcqInfo.create(acq=acq)
+    if atype == "hfb" and ftype.name == "hfb":
+        if not acq.corrinfos.count():
+            try:
+                di.CorrAcqInfo.create(
+                    acq=acq, **get_acqcorrinfo_keywords_from_h5(fullpath)
+                )
+                log.info(
+                    'Added information for correlator acquisition "%s" to '
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py]:
```suggestion
                    'Added information for HFB acquisition "%s" to '
```
"Correlator" refers to the `corr` file type in this case

File: alpenhorn/auto_import.py [hunk 2]
```diff
@@ -470,6 +533,22 @@ def _import_file(node, root, acq_name, file_name):
                     "empty. Leaving fields NULL." % (acq_name)
                 )
                 di.CorrAcqInfo.create(acq=acq)
+    if atype == "hfb" and ftype.name == "hfb":
+        if not acq.corrinfos.count():
+            try:
+                di.CorrAcqInfo.create(
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py]:
```suggestion
                di.HFBAcqInfo.create(
```
(and throughout).  The `CorrAcqInfo` table is for files of type `corr`.

File: alpenhorn/auto_import.py [hunk 3]
```diff
@@ -470,6 +533,22 @@ def _import_file(node, root, acq_name, file_name):
...
+                log.info(
+                    'Added information for correlator acquisition "%s" to '
+                    "DB." % acq_name
+                )
+            except:
+                log.warning(
+                    'Missing info for acquistion "%s": HDF5 datasets '
+                    "empty. Leaving fields NULL." % (acq_name)
+                )
+                di.CorrAcqInfo.create(acq=acq)
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py]:
```suggestion
                di.HFBAcqInfo.create(acq=acq)
```

File: alpenhorn/auto_import.py [hunk 4]
```diff
@@ -239,6 +284,24 @@ def get_filecorrinfo_keywords_from_h5(path):
         "freq_number": freq_number,
     }
 
+    
+def get_filehfbinfo_keywords_from_h5(path):
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py:288]: This function is never called.

You probably intended to call it before creating a new `HKPFileInfo` entry as part of the big if-elif clause down around here https://github.com/chime-experiment/alpenhorn/blob/e4685ef8dd42040f49960edc8f84ba1ee8cce7dd/alpenhorn/auto_import.py#L810

File: alpenhorn/auto_import.py [hunk 5]
```diff
@@ -197,6 +197,51 @@ def get_acqcorrinfo_keywords_from_h5(path):
     f.close()
     return {"integration": integration, "nfreq": n_freq, "nprod": n_prod}
 
+def get_acqhfbinfo_keywords_from_h5(path):
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py:200]: This function is never called.

Presumably you want to call it before creating the new `HFBAcqInfo` entry below to get the data to populate the new record with.

Down here: https://github.com/chime-experiment/alpenhorn/blob/e4685ef8dd42040f49960edc8f84ba1ee8cce7dd/alpenhorn/auto_import.py#L536

File: alpenhorn/auto_import.py [hunk 6]
```diff
@@ -470,6 +533,22 @@ def _import_file(node, root, acq_name, file_name):
                     "empty. Leaving fields NULL." % (acq_name)
                 )
                 di.CorrAcqInfo.create(acq=acq)
+    if atype == "hfb" and ftype.name == "hfb":
```
Comment (ketiltrout) [inline alpenhorn/auto_import.py]:
```suggestion
    elif atype == "hfb" and ftype.name == "hfb":
```
This is part of one big if-elif statement running though all the acq types

**Review by ketiltrout — CHANGES_REQUESTED** [review]
There seems to be confusion here over whether HFB files are `corr` files or something new.

You're adding entries to `CorrAcqInfo` here but over in https://github.com/chime-experiment/chimedb_di/pull/14 you've defined new HFB-specific tables.

So, definitely step one is to figure out whether HFB files are `corr` files or not.

If they're not, you'll also need to define them in the acq_format file before this PR can be approved.  The TeX for that is over here and the document provides instructions on the procedure for updating itself: https://github.com/chime-experiment/ch_memos/blob/master/acq_format/acq_format.tex

The PDF version of the acq format document is in doclib: https://bao.chimenet.ca/doc/documents/5

**Comment by jrs65** [comment]
@james-s-willis as Don says you'll also need to create a new hfb acq type for this. The corr acq type is to hold visibility correlation (which these are not).

**Comment by james-s-willis** [comment]
https://github.com/chime-experiment/alpenhorn/blob/2796cfd97db32346fd4a00132f57f6e5cda87cea/alpenhorn/auto_import.py#L589
I know I'm not editing this part, but shouldn't this line have `file.hkinfos.count()` instead of `corrinfos`? @ketiltrout

**Comment by james-s-willis** [comment]
@ketiltrout, @jrs65 I have updated this PR with your corrections and I have created a new PR (https://github.com/chime-experiment/ch_memos/pull/4) for the HFB acq type in the CHIME Archive Data Format document.

**Comment by ketiltrout** [comment]
> I know I'm not editing this part, but shouldn't this line have `file.hkinfos.count()` instead of `corrinfos`? @ketiltrout

More amazingness from this code.  You're right, and I made an issue for it, but really it looks like that `if` should be deleted and the `except` line above should be fixed so it only catches the exception thrown if the update fails, not everything.

**Review by ketiltrout — APPROVED** [review]

**Pull request merged**